packages/diagrams/lib/src/custom_list_item.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:flutter/material.dart';
+import 'diagram_step.dart';
+
+class _VideoDescription extends StatelessWidget {
+  const _VideoDescription({
+    Key key,
+    this.title,
+    this.user,
+    this.viewCount,
+  }) : super(key: key);
+
+  final String title;
+  final String user;
+  final int viewCount;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(5.0, 0.0, 0.0, 0.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Text(
+            title,
+            style: TextStyle(
+              fontWeight: FontWeight.w500,
+              fontSize: 14.0,
+            ),
+          ),
+          const Padding(padding: EdgeInsets.symmetric(vertical: 2.0)),
+          Text(user, style: TextStyle(fontSize: 10.0)),
+          const Padding(padding: EdgeInsets.symmetric(vertical: 1.0)),
+          Text(
+            '$viewCount views',
+            style: TextStyle(fontSize: 10.0),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// A sample list item that looks similar to a YouTube related video item.
+///
+/// Since users are misusing ListTile material widget, this example was created
+/// to help provide inspiration for alternative ways to create list items.
+class CustomListItem extends StatelessWidget {
+  const CustomListItem({
+    this.user,
+    this.viewCount,
+    this.title,
+  });
+
+  final String user;
+  final int viewCount;
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 5.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          const Expanded(child: FlutterLogo(size: 96), flex: 2),
+          Expanded(
+            child: _VideoDescription(
+              title: title,
+              user: user,
+              viewCount: viewCount,
+            ),
+            flex: 3,
+          ),
+          const Icon(Icons.more_vert, size: 16.0),
+        ],
+      ),
+    );
+  }
+}
+
+class CustomListItemDiagram extends StatelessWidget
+    implements DiagramMetadata {
+  const CustomListItemDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  Widget build(BuildContext context) {
+    return ConstrainedBox(
+      key: UniqueKey(),
+      constraints: BoxConstraints.tight(const Size(400.0, 250.0)),
+      child: Container(
+        alignment: FractionalOffset.center,
+        padding: const EdgeInsets.all(5.0),
+        color: Colors.white,
+        child: ListView.builder(
+          padding: const EdgeInsets.all(8.0),
+          itemExtent: 106.0,
+          itemBuilder: (BuildContext context, int index) {
+            return const CustomListItem(
+              user: 'Flutter',
+              viewCount: 999000,
+              title: 'The Flutter YouTube Channel is Here!',
```

Comment:
Just aesthetically, it would be nice if 1) this label didn't wrap in the tile, and 2) there were different labels on different items, so it looks more like a "real" example, and not just a test image.